packages/joy-roles/src/index.tsx

```diff
@@ -27,95 +27,60 @@ type Props = AppProps & ApiProps & I18nProps & {
   allAccounts?: SubjectInfo,
 };
 
-type State = {
-  tabs: Array<TabItem>,
-};
-
-class App extends React.PureComponent<Props, State> {
-  state: State;
-  transport: ITransport
-  mockTransport: ITransport
-  oppCtrl: OpportunitiesController
-  applyCtrl: ApplyController
-  myRolesCtrl: MyRolesController
-
-  constructor(props: Props) {
-    super(props);
-
-    this.transport = new Transport(props)
-    this.mockTransport = new MockTransport()
-    this.oppCtrl = new OpportunitiesController(this.mockTransport)
-    this.applyCtrl = new ApplyController(this.mockTransport)
-    this.myRolesCtrl = new MyRolesController(this.mockTransport)
-
-    const { t } = props;
-
-    this.state = {
-      tabs: [
-        {
-          isRoot: true,
-          name: 'working-groups',
-          text: t('Working groups')
-        },
-        {
-          name: 'opportunities',
-          text: t('Opportunities')
-        },
-        {
-          name: 'my-roles',
-          text: t('My roles')
-        },
-      ],
-    };
+export const App: React.FC<Props> = React.memo((props: Props) => {
+  const { t } = props
+  const tabs = [
+    {
+      isRoot: true,
+      name: 'working-groups',
+      text: t('Working groups')
+    },
+    {
+      name: 'opportunities',
+      text: t('Opportunities'),
+      hasParams: true,
+    },
+    {
+      name: 'my-roles',
+      text: t('My roles')
+    },
+  ]
+
+  //const transport = new Transport(props)
+  const mockTransport = new MockTransport()
```

Comment:
Why not to call this variable just `transport`. Then you don't need to refactor that name on the other lines of this file where you use `mockTransport` now.
```suggestion
  const transport = new MockTransport()
```

Reply:
Because we will use both. I commented this out to quickly move on to another refactor; it will make sense in the next PR.